packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -147,19 +188,92 @@ export const createSchema = (schema: Schema, config?: DataSourceGenerateConfig):
   }
 
+  const engine = schema.getEngine().type;
+
+  const nullEnums = schema  // find null enums in all models
+    .getModels()
+    .map((model) => model
+        .getFields()
+        .filter((field) => field.type.kind === 'Enum')
+        .map((field) => {
+          if(field.type.kind === 'Enum'){
+            return createEnums(field.type, engine, model.getName(), field.name);
+          }
+          else{
+            return undefined;
+          }
+     }));
+
+    const notNullEnums = schema  // find Non null enums in all models
+    .getModels()
+    .map((model) => model
+        .getFields()
+        .filter((field) => field.type.kind === 'NonNull' && field.type.type.kind === 'Enum')
+        .map((field) => {
+          if(field.type.kind === 'NonNull' && field.type.type.kind === 'Enum'){
+            return createEnums(field.type.type, engine, model.getName(), field.name);
+          }
+          else{
+            return undefined;
+          }
+     }));
+
   const models = schema
     .getModels()
     .filter((model) => model.getPrimaryKey())
     .map((model) => {
-      return createModel(model);
+      return createModel(model, engine);
     });
+
+  const combinedEnums = nullEnums.concat(notNullEnums).flat() // making 1 D array
+
+  if (engine === 'Postgres') { // for postgresql
+    const seenEnums = new Set<string>();
+    const uniqueEnums = combinedEnums.filter((node) => {
+      if(seenEnums.has(node['name'].escapedText)) {
+        return false;
+      }
+      else{
+        seenEnums.add(node['name'].escapedText);
+        return true;
+      }
+    });
+    const modelsWithEnums = models.concat(uniqueEnums); // appending enums to models
+  
+    const tsSchema = ts.factory.createCallExpression(
+      ts.factory.createPropertyAccessExpression(
+        createConfigureExpression(schema, config),
+        ts.factory.createIdentifier(TYPESCRIPT_DATA_SCHEMA_CONSTANTS.SCHEMA_METHOD), // create a.schema()
+      ),
+      undefined,
+      [ts.factory.createObjectLiteralExpression(modelsWithEnums as ts.ObjectLiteralElementLike[], true)],
+    );
+    return ts.factory.createVariableStatement(
+      [exportModifier],
+      ts.factory.createVariableDeclarationList(
+        [
+          ts.factory.createVariableDeclaration(
+            ts.factory.createIdentifier(TYPESCRIPT_DATA_SCHEMA_CONSTANTS.EXPORT_VARIABLE_NAME),
+            undefined,
+            undefined,
+            tsSchema,
+          ),
+        ],
+        ts.NodeFlags.Const,
+      ),
+    );
+  } 
+
+    
+  const modelsWithEnums = models.concat(combinedEnums); // appending enums to models for MySql
+  
   const tsSchema = ts.factory.createCallExpression(
     ts.factory.createPropertyAccessExpression(
       createConfigureExpression(schema, config),
-      ts.factory.createIdentifier(TYPESCRIPT_DATA_SCHEMA_CONSTANTS.SCHEMA_METHOD),
+      ts.factory.createIdentifier(TYPESCRIPT_DATA_SCHEMA_CONSTANTS.SCHEMA_METHOD), // create a.schema()
```

Comment:
prefer to remove dead code rather than leaving it commented
```suggestion
      ts.factory.createIdentifier(TYPESCRIPT_DATA_SCHEMA_CONSTANTS.SCHEMA_METHOD), 
```

Reply:
It isn't a dead code its a comment, but its not very necessary since it is existing code so will probably remove it